webapp/views/student.py

```diff
@@ -93,6 +134,50 @@ def submissions(student: Student | None, page: int):
     )
 
 
+@blueprint.route("/home", methods=["GET"])
+@authorize(db.students)
+def home(student: Student | None):
+    if config.config.registration and not student:
+        return redirect("/login")
+    if config.config.registration and student and student.group is None:
+        return redirect("/")
+    if config.config.registration and student and student.group is not None and student.variant is None:
+        return redirect("/variant")
+    if config.config.exam:
+        return redirect("/exam")
+    greeting_message = get_greeting_msg()
+    hide_pending = config.config.exam and request.args.get('hide_pending', False)
+    group = statuses.get_group_statuses(student.group, hide_pending)
+    tasks_statuses = list(int(task_status.status) for task_status in group.variants[student.variant].statuses)
+    group_place = home_manager.get_group_place(student.group)
+    student_place = home_manager.get_student_place(student.group, student.variant)
+    variants = db.variants.get_all()
+    return render_template(
+        "student/home.jinja",
+        student=student,
+        greeting_message=greeting_message,
+        registration=config.config.registration,
+        group_rating=config.config.groups,
+        exam=config.config.exam,
+        group=group,
+        variant=student.variant,
+        variants=variants,
+        number_of_tasks=len(group.variants),
+        tasks_statuses=tasks_statuses,
+        group_place=group_place,
+        student_place=student_place,
+    )
```

Comment:
Есть мнение, что этот фрагмент кода можно упростить:
1. Можно избавиться от `hide_pending`, эта функциональность нужна преподавателям в режиме зачёта или пересдач, чтобы скрывать варианты без зачтённых решений и не листать вверх-вниз, когда сдаёт поток. На домашней странице такая функциональность необязательна.
2. Можно заинлайнить ряд переменных, которые используются только в `render_template`.
```suggestion
    group = statuses.get_group_statuses(student.group, False)
    tasks_statuses = list(ts.status for ts in group.variants[student.variant].statuses)
    return render_template(
        "student/home.jinja",
        student=student,
        greeting_message=get_greeting_msg(),
        registration=config.config.registration,
        group_rating=config.config.groups,
        exam=config.config.exam,
        group=group,
        variant=student.variant,
        variants=db.variants.get_all(),
        number_of_tasks=len(group.variants),
        tasks_statuses=tasks_statuses,
        group_place=home_manager.get_group_place(student.group),
        student_place=home_manager.get_student_place(student.group, student.variant),
    )
```

Reply:
Спасибо за замечание! Исправил в [fix: optimized code](https://github.com/kispython-ru/dta/pull/56/commits/f26d8c3f35e95cde9311dba7e6b71d362b5668d1)